hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -401,13 +441,25 @@ protected void completeClustering(HoodieReplaceCommitMetadata metadata, JavaRDD<
   @Override
   public void syncTableMetadata() {
     // Open up the metadata table again, for syncing
-    try (HoodieTableMetadataWriter writer = SparkHoodieBackedTableMetadataWriter.create(hadoopConf, config, context)) {
+    try {
+      HoodieTableMetadataWriter writer =
```

Comment:
we lose auto closing by moving to  a regular try-catch. Why is this change needed?